src/Propel/Generator/Manager/DataDictionaryExportManager.php

```diff
@@ -0,0 +1,262 @@
+## Table **$tableName** $flags
+<a name="$anchorName"></a>
+$descriptionMd
+### Columns:
+|$columnsHeaderMd|
+$headerSepMd
+$columnRowsMd
+
+[to table of contents](#TOC)
+EOT;
+    }
+
+    /**
+     * @param \Propel\Generator\Model\Table $table
+     *
+     * @return string
+     */
+    protected function getTableFlags(Table $table): string
+    {
+        $isAbstract = $table->isAbstract();
+        $skipSql = $table->isSkipSql();
+
+        if (!$isAbstract && !$skipSql) {
+            return '';
+        }
+        $flags = [];
+
+        if ($isAbstract) {
+            $flags[] = 'abstract';
+        }
+        if ($skipSql) {
+            $flags[] = 'skip SQL';
+        }
+
+        return '*(' . implode(', ', $flags) . ')*';
+    }
+
+    /**
+     * @param \Propel\Generator\Model\Column $column
+     *
+     * @return string
+     */
+    protected function buildMarkdownForColumn(Column $column): string
+    {
+        $defaultValueString = $column->getDefaultValueString();
+        $defaultValue = ($defaultValueString === 'null') ? '' : $defaultValueString;
+
+        $columnRow = [
+            $column->getName(),
+            $column->getPhpName(),
+            $column->getType(),
+            $column->getSize(),
+            $this->getFlagSymbol($column->isPrimaryKey()),
+            $this->getFlagSymbol($column->isAutoIncrement()),
+            $this->getFlagSymbol($column->isNotNull()),
+            $this->getFlagSymbol($column->isUnique()),
+            $this->buildLinksToForeignTables($column),
+            $defaultValue,
+            $column->getDescription(),
+        ];
+
+        return '|' . implode('|', $columnRow) . '|';
+    }
+
+    /**
+     * @param \Propel\Generator\Model\Column $column
+     *
+     * @return string
+     */
+    protected function buildLinksToForeignTables(Column $column): string
+    {
+        $foreignKeys = $column->getForeignKeys();
+        if (count($foreignKeys) === 0) {
+            return '';
+        }
+
+        $links = [];
+        foreach ($foreignKeys as $fk) {
+            $foreignTableName = $fk->getForeignTableName();
+            $links[] = $this->buildLink($foreignTableName);
+        }
+
+        return implode(' ', $links);
+    }
+
+    /**
+     * @param string $markdownContent
+     * @param string $baseFilename
+     *
+     * @return void
+     */
+    protected function writeFile(string $markdownContent, string $baseFilename): void
+    {
+        $file = $this->getWorkingDirectory() . DIRECTORY_SEPARATOR . $baseFilename . '.schema.md';
+
+        $this->log('Writing md file to ' . $file);
+
+        file_put_contents($file, $markdownContent);
+    }
+
+    /**
+     * @param bool $val
+     *
+     * @return string
+     */
+    protected function getFlagSymbol(bool $val): string
+    {
+        return ($val) ? '*' :  '';
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return string
+     */
+    private function toAlphnum(string $name): string
```

Comment:
maybe we can use real words here? also protected instead of private maybe.

Reply:
Ok, changed it